liblangutil/ErrorReporter.h

```diff
@@ -87,6 +87,34 @@ class ErrorReporter
 
 	void typeError(SourceLocation const& _location, std::string const& _description);
 
+	template <typename... Strings>
+	void typeErrors(SourceLocation const& _location, Strings const&... _descriptions)
+	{
+		std::initializer_list<std::string> const _descs = {_descriptions...};
+		std::string errorStr;
```

Comment:
```suggestion
		std::string errorStr = joinHumanReadable(_descs, " ");
```

Reply:
`joinHumanReadable()` is not smart enough for this case. The problem is that we have often empty strings that we concatenate, e.g. when `isImplicitlyConvertibleTo()` returns an empty error.

Example:

```
auto str1 = "This is wrong.";
auto str2 = "";

auto str = joinHumanReadable(str1, str2, " ");
// Str would now be:  "This is wrong. " (trailing space)
```

Reply:
Ah right, I see, we talked about that. Hm.

Reply:
But probably still better to extract this into a helper function

Reply:
It might make sense to extend `joinHumanReadable()` itself though. I mean is there ever a case where we'd want that with any other string? I believe that would almost always be wrong.

Reply:
I found this solution now:
```
		auto filterEmpty = boost::adaptors::filtered([](std::string s) { return s.size() > 0; });

		std::initializer_list<std::string> const descs = {_descriptions...};
		std::string errorStr = dev::joinHumanReadable(descs | filterEmpty);
```